### ISSUE TITLE
[arm64] Assertion failed 'emitCurIG != emitPrologIG'

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#14568]
Found in an x64_arm64_altjit JitStress=2 JitStressRegs=1 run.

E.g.,
```
Assertion failed 'emitCurIG != emitPrologIG' in 'DomainBoundILStubClass:IL_STUB_PInvoke(long,struct,double,struct,float,struct,int,struct,char,struct,int,struct,float,struct,long,struct,float,struct,double,struct,float,struct,short,struct):struct' (IL size 262)

    File: e:\gh\coreclr2\src\jit\emit.cpp Line: 6634
    Image: E:\gh\coreclr2\bin\tests\Windows_NT.x64.Checked\Tests\Core_Root\CoreRun.exe
```

Tests:
```
JIT\jit64\mcc\interop\mcc_i77\mcc_i77.cmd
JIT\jit64\mcc\interop\mcc_i76\mcc_i76.cmd
```

**Comment by BruceForstall** [comment]
@dotnet/arm64-contrib

**Comment by BruceForstall** [comment]
I see this in JitStress=2, JitStressRegs=4 in CI:

https://ci.dot.net/job/dotnet_coreclr/job/master/view/arm64/job/jitstress/job/arm64_cross_checked_windows_nt_jitstress2_jitstressregs4_tst/20/artifact/bin/tests/Windows_NT.arm64.Checked/Smarty.run.0/Smarty.0.fail.smrt/*view*/

```
Tests.lst=mcc_i75.cmd_5229, , , Smrt00000001, , # JIT\jit64\mcc\interop\mcc_i75\mcc_i75.cmd  CATS: EXPECTED_PASS;Pri1
Tests.lst=mcc_i74.cmd_5228, , , Smrt00000001, , # JIT\jit64\mcc\interop\mcc_i74\mcc_i74.cmd  CATS: EXPECTED_PASS;Pri1
Tests.lst=mcc_i76.cmd_5230, , , Smrt00000001, , # JIT\jit64\mcc\interop\mcc_i76\mcc_i76.cmd  CATS: EXPECTED_PASS
Tests.lst=mcc_i77.cmd_5231, , , Smrt00000001, , # JIT\jit64\mcc\interop\mcc_i77\mcc_i77.cmd  CATS: EXPECTED_PASS
```

**Comment by sdmaclea** [comment]
Those are var arg tests which are not supported on linux.  So in general I have not been running them.

**Comment by AndyAyersMS** [comment]
This may have been fixed by #17845.

@briansull can you take a look?

**Comment by briansull** [comment]
Fixed with #17845
